invoices.go

```diff
@@ -209,9 +213,46 @@ func (i *InvoiceCollection) UnmarshalXML(d *xml.Decoder, start xml.StartElement)
 		Transactions:            v.ChargeInvoice.Transactions,
 		CreditPayments:          v.ChargeInvoice.CreditPayments,
 	}
+	var creditInvoices []Invoice
+	for _, ci := range v.CreditInvoices {
+		creditInvoices = append(creditInvoices, Invoice{
```

Comment:
This block is essentially doing the same thing as the block above. It'll be easy to introduce bugs where we add a field to the invoice and forget to add it to this block.

Instead let's create function to do the conversion and use that. Maybe something like this:

```go
func (i invoiceFields) ToInvoice() *Invoice {
   // ...
}
```

Then here can can just do

`creditInvoices[i] = ci.invoiceFields.ToInvoice()`

And use the same conversion above.

Reply:
done